sys/include/net/cord/lc.h

```diff
@@ -0,0 +1,213 @@
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_cord_lc CoRE RD Lookup Client
+ * @ingroup     net_cord
+ * @brief       Library for using RIOT as CoRE Resource Directory (RD) lookup client
+ *
+ * This module implements a CoRE Resource Directory lookup client library, that
+ * allows RIOT nodes to lookup resources, endpoints and groups with resource
+ * directories.
+ * It implements the standard lookup functionality as defined in
+ * draft-ietf-core-resource-directory-23.
+ * @see https://tools.ietf.org/html/draft-ietf-core-resource-directory-23
+ *
+ * ## Lookup modes
+ *
+ * The module defines two types of lookup for interacting with a RD server:
+ *
+ * - raw: result of the lookup is returned as is. No `page` or `count` filter
+ * is applied by default. Use @ref cord_lc_raw() for this mode.
+ * - pre-parsed: result of the lookup is parsed and returned in a
+ * @ref cord_lc_res_t or @ref cord_lc_ep_t depending on the type of the
+ * lookup. The default `count` filter is set to `1` and `page` filter is
+ * incremented after each successful call and resets to `0` when lookup result
+ * is empty. Use @ref cord_lc_res() or cord_lc_ep() for this mode.
+ *
+ * ## Limitations
+ *
+ * Currently, this module cannot do more than a single request concurrently
+ * and the request is fully synchronous. The client can only connects to one
+ * RD server at a time. The client will disconnect when a connection to a new
+ * RD server is made, regardless of whether the connection attempt is successful
+ * or not.
+ *
+ * @{
+ *
+ * @file
+ * @brief       CoRE Resource Directory lookup interface
+ *
+ * @author      Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ */
+
+#ifndef NET_CORD_LC_H
+#define NET_CORD_LC_H
+
+#include "net/sock/udp.h"
+#include "net/nanocoap.h"
+#include "clif.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Return values and error codes used by this module
+ */
+enum {
+    CORD_LC_OK        =  0,     /**< everything went as expected */
+    CORD_LC_TIMEOUT   = -1,     /**< no response from the network */
+    CORD_LC_ERR       = -2,     /**< internal error or invalid reply */
+    CORD_LC_OVERFLOW  = -3,     /**< internal buffers can not handle input */
+    CORD_LC_NORSC     = -4,     /**< lookup interface not found */
+};
+
+/**
+ * @brief  RD lookup types
+ */
+enum {
+    CORD_LC_RES,                /**< Resource lookup type */
+    CORD_LC_EP,                 /**< Endpoint lookup type */
+};
+
+/**
+ * @brief  Information about RD server and its lookup interface resources
+ */
+typedef struct {
+    const sock_udp_ep_t *remote;        /**< Remote endpoint of RD server */
+    char *res_lookif;                   /**< Resource lookup interface */
+    size_t res_lookif_len;              /**< Length of @p res_lookif */
+    char *ep_lookif;                    /**< Endpoint lookup interface */
+    size_t ep_lookif_len;               /**< Length of @p ep_lookif */
```

Comment:
looking below these should be `\0` terminated strings to be used as path in `gcoap_req_init`, hence you can omit the length attribute and the memcpy below